pkg/crypto.go

```diff
@@ -616,3 +616,34 @@ func (client *Crypto) encryptPlainTextWith(plaintext []byte, key *rsa.PublicKey)
 	}
 	return ciphertext, nil
 }
+
+func serialNumber() (int64, error) {
+	// TODO: Make this implementation safer. This one just hopes for enough entropy.
+	n, err := rand.Int(rand.Reader, big.NewInt(time.Now().UnixNano()*2))
```

Comment:
```suggestion
	serialNumberLimit := new(big.Int).Lsh(big.NewInt(1), 128)
	serialNumber, err := rand.Int(rand.Reader, serialNumberLimit)
```
as used by https://golang.org/src/crypto/tls/generate_cert.go

Reply:
Good find :+1: